src/custom_hooks/useUrlParams.js

```diff
@@ -0,0 +1,18 @@
+import React from "react";
+
+export function useUrlParams() {
+    const [url_params, setUrlParams] = React.useState([])
+
+    React.useEffect(() => {
+        let params_array = [];
+        if (window.location.search) {
+            const params = new URLSearchParams(window.location.search)
+            for (const param of params) {
+                params_array = [...params_array, param];
+            }
+            setUrlParams(params_array);
+        }
+    }, [window.location.search]);
```

Comment:
I'm wandering if it is a good practice to add window.location in useEffect depth array.
see please this link, maybe we will be able to improve a little bit this part of code:
https://stackoverflow.com/questions/58442168/why-useeffect-doesnt-run-on-window-location-pathname-changes

Reply:
Thank you, I was not aware that window location can cause issues. I will find a better solution for this.